File: MyProject/Source/MyProject/Private/BasicFrame/Libs/Task/TaskQueue.cpp
#include "MyProject.h"
#include "TaskQueue.h"
#include "ITask.h"
#include "TaskThreadPool.h"

TaskQueue::TaskQueue(std::string name)
    : Super(name)
{
	mTaskThreadPool = new TaskThreadPool();
}

void TaskQueue::push(ITask item)
{
	Super::push(item);

    // 检查是否有线程空闲，如果有就唤醒
    mTaskThreadPool->notifyIdleThread();
}

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/Task/TaskQueue.cs
namespace SDK.Lib
{
    public class TaskQueue : MLockQueue<ITask>
    {
        public TaskThreadPool mTaskThreadPool;

        public TaskQueue(string name)
            : base(name)
        {

        }

        override public void init()
        {

        }

        override public void dispose()
        {

        }

        public new void push(ITask item)
        {
            base.push(item);

            // 检查是否有线程空闲，如果有就唤醒
            this.mTaskThreadPool.notifyIdleThread();
        }
    }
}